lib/azureServiceClient.ts

```diff
@@ -41,6 +41,13 @@ export class AzureServiceClient extends ServiceClient {
   constructor(credentials: ServiceClientCredentials, options?: AzureServiceClientOptions) {
     super(credentials, options = updateOptionsWithDefaultValues(options));
 
+    // For convenience, if the credentials have an associated AzureEnvironment,
+    // automatically use the baseUri from that environment.
+    const env = (credentials as any).environment;
```

Comment:
Can you add a type for `env`? I have no idea what it is right now.

Reply:
The type lives in `ms-rest-nodeauth` :( We could double-define it? Or migrate it over here?

Reply:
We're definitely taking a dependency on the type's structure here (We want an object that has a `resourceManagerEndpointUrl` string property on it). I don't like the idea of double-defining the type. ms-rest-nodeauth relies on ms-rest-js and ms-rest-azure-env has a lot of Azure specific stuff in it, even though it doesn't rely on ms-rest-azure-js. This looks like it could be simplified a lot.
Now that I understand this a little better I'd recommend not doing anything here yet. I'll create an issue for re-thinking how we structure these repositories and we can all talk about it and plan if/when we should do something about it.